### PR TITLE
Add-on: Signup flow, implement step on Free sites to sell add-ons

### PR DESCRIPTION
**Pull request by claudiucelfilip**
#### Proposed Changes

* Adds the add-ons step to the signup flow if a Free plan is selected.
* Changes are hidden behind the `signup/add-ons` flag for all environments.
* Add-on cards are reused from #64289 

Addresses #64121 

pricing - white version

#### Testing Instructions

* Go to `/start?flags=signup/add-ons`
* Select a Free plan
* You should see the Addons step
* Select a few add-ons and continue
* You should see the addons in your Shopping cart, on the Checkout page

* Selecting a Paid plan should not show you the Add-ons step

#### TODO
- [x] Replace subtitle copy placeholder
- [x] Show `Added to your plan` instead of `Included in your plan`
- [x] Show `Popular` label next to Premium themes

### CODE REVIEW

File: client/my-sites/add-ons/utils/get-feature-slug.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		case PRODUCT_WPCOM_CUSTOM_DESIGN:
+			return WPCOM_FEATURES_CUSTOM_DESIGN;
+		case PRODUCT_NO_ADS:
+			return WPCOM_FEATURES_NO_ADVERTS;
+		default:
+			return null;
+	}
+};
+
+export default getFeatureSlug;
```
Comment (chriskmnds) [inline client/my-sites/add-ons/utils/get-feature-slug.ts]: hmm I think something to do with the rebase here. this file/folder were deleted

File: client/signup/steps/add-ons/index.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	flowName: string;
+	positionInFlow: number;
+	defaultDependencies: object;
+	forwardUrl: string;
+}
+interface AddOnsProps {
+	selectedAddOns: string[];
+	addOns: ( AddOnMeta | null )[];
+	onToggleAllAddOns: () => void;
+	onAddAddon: ( addonSlug: string ) => void;
```
Comment (chriskmnds) [inline client/signup/steps/add-ons/index.tsx]:
```suggestion
	onAddAddOn: ( addOnSlug: string ) => void;
```

Nitpick - no need to update. It happens in a few places, but mainly curious - should we camelCase `addon` -> `addOn`? Not sure if I'm getting it wrong instead! 😬

**Review by chriskmnds — APPROVED** [review]
LGTM. 

Some minor points I noticed below, but things we can address in follow-ups (if issues at all):

1. Some padding/margin at the bottom of the grid container might be good when scrolling to the bottom:

Screenshot 2022-06-30 at 3 23 15 PM

2. I see there is always a `Step x of 3` top-right even when only two steps are effective with no option for a free plan. Not sure if we need to bother with this (or if anything we can even do about it).

Screenshot 2022-06-30 at 3 22 36 PM

3. When on checkout the add-ons show different meta under the title when it seems they all could be the same i.e. not sure why the difference matters.

Screenshot 2022-06-30 at 3 19 21 PM

cc @gmovr (in case any worth addressing in follow-ups)

File: client/layout/index.jsx [hunk 3]
```diff
@@ -47,6 +47,8 @@ import { handleScroll } from './utils';
 // goofy import for environment badge, which is SSR'd
 import 'calypso/components/environment-badge/style.scss';
 import './style.scss';
+import { ContextExclusionPlugin } from 'webpack';
+import { combineXPosts } from 'calypso/state/reader/streams/utils';
```
Comment (chriskmnds) [inline client/layout/index.jsx]: These are leftover imports to cleanup I guess 🤔

**Comment by chriskmnds** [comment]
Just noticed a bit of a bug/issue: 

If you select an add-on and go to checkout, then going back to the flow will not let you add other add-ons to your cart, until the previous one is "removed" from cart on the checkout page. The previously selected add-on that lingers in the cart is not shown as selected on the add-ons page either. This happened to me with any add-on combination.

Kapture 2022-07-05 at 15 10 15

Reproduce:

1. go to `start/add-ons`
2. select "Premium Themes" add-on and go to checkout. do not complete the purchase
3. go back to `start/add-ons` (any way)
4. select "Custom CSS" add-on and go to checkout.
5. "Premium Themes" should still show for checkout (and no "Custom CSS").

@gmovr can you confirm if this is reproducible on your end pls?

**Comment by gmovr** [comment]
> Just noticed a bit of a bug/issue:
> 
> If you select an add-on and go to checkout, then going back to the flow will not let you add other add-ons to your cart, until the previous one is "removed" from cart on the checkout page. The previously selected add-on that lingers in the cart is not shown as selected on the add-ons page either. This happened to me with any add-on combination.
> 
> Kapture 2022-07-05 at 15 10 15 [ ](https://user-images.githubusercontent.com/1705499/177326526-7d633e14-967f-4a09-966b-879687faa044.gif) Kapture 2022-07-05 at 15 10 15 [ ](https://user-images.githubusercontent.com/1705499/177326526-7d633e14-967f-4a09-966b-879687faa044.gif)
> 
> Reproduce:
> 
> 1. go to `start/add-ons`
> 2. select "Premium Themes" add-on and go to checkout. do not complete the purchase
> 3. go back to `start/add-ons` (any way)
> 4. select "Custom CSS" add-on and go to checkout.
> 5. "Premium Themes" should still show for checkout (and no "Custom CSS").
> 
> @gmovr can you confirm if this is reproducible on your end pls?

I'm seeing the same -- I will make a card for this and do some additional testing.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7419599

Thank you @claudiucelfilip for including a screenshot in the description! This is really helpful for our translators.

**Comment by DavidRothstein** [comment]
>When on checkout the add-ons show different meta under the title when it seems they all could be the same i.e. not sure why the difference matters.
>
>Screenshot 2022-06-30 at 3 19 21 PM

I noticed that issue independently, and have a quick fix for it at https://github.com/Automattic/wp-calypso/pull/65315.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7419599).

**Pull request merged**